crate_universe/private/generate_utils.bzl

```diff
@@ -205,16 +205,14 @@ def generate_config(repository_ctx):
     annotations = collect_crate_annotations(repository_ctx.attr.annotations, repository_ctx.name)
 
     # Load additive build files if any have been provided.
-    content = list()
     for data in annotations.values():
-        additive_build_file_content = data.pop("additive_build_file_content", None)
-        if additive_build_file_content:
-            content.append(additive_build_file_content)
-        additive_build_file = data.pop("additive_build_file", None)
-        if additive_build_file:
-            file_path = repository_ctx.path(Label(additive_build_file))
-            content.append(repository_ctx.read(file_path))
-        data.update({"additive_build_file_content": "\n".join(content) if content else None})
+        f = data.pop("additive_build_file", None)
+        content = "\n".join([
+            data.pop("additive_build_file_content", None) or "",
+            repository_ctx.read(Label(f)) if f else "",
+        ]).strip()
```

Comment:
Since this content is generally rendered at the bottom of a file, could we avoid the `strip` so we don't strip the final newline?

Reply:
Hmm, good point. The strip was there to prevent the content from being just `\n`, as well as keeping the source a bit cleaner. Would it make sense to explicitly add a newline in addition to stripping? Since there is no guarantee of a newline from `repository_ctx.read` anyway.

Reply:
I lean toward the more hands off approach where we just render what users told us. That way folks who care about newlines have the power to fix it

Reply:
I agree, that makes sense. Would you mind taking another look?